pypeit/core/trace_slits.py

```diff
@@ -37,6 +39,57 @@
 import time
 
 
+def extrapolate_trace(traces_in, spec_min_max_in):
+    """
+    Linearly extrapolates trace to fill in pixels that lie outside of the range spec_min, spec_max). This
+    routine is useful for echelle spectrographs where the orders are shorter than the image by a signfiicant
+    amount, since the polynomial trace fits often go wild.
+
+    Args:
+        traces (np.ndarray): Tuple or array with size (nspec,) or (nspec, ntrace) containing object or slit boundary traces
+        spec_min_max: Minimum and maximum pectral pixel for which trace is valid.  If this is a 2-d array,
+        the same numbers will be used for all traces in traces_in.  If an array, then this must be an ndarray of shape
+        (2, ntrace,) where the spec_min_max[0,:] are the minimua and spec_min_max[1,:] are the maxima.
+
+    Returns:
+        trace_extrap (np.ndarray): Array with same size as trace containing the linearly extrapolated values for the bad
+        spectral pixels.
+    """
+
+    # This little bit of code allows the input traces to either be (nspec, nslit) arrays or a single
+    # vectors of size (nspec)
+    spec_min_max_tmp = np.array(spec_min_max_in)
+    if traces_in.ndim == 2:
+        traces = traces_in
+        nslits = traces.shape[1]
+        if np.array(spec_min_max_in).ndim == 1:
+            spec_min_max = np.outer(spec_min_max_tmp, np.ones(nslits))
+        elif spec_min_max_tmp.ndim == 2:
+            if (spec_min_max_tmp.shape[1] != nslits):
+                msgs.error('If input as any arrays, spec_min_max needs to have dimensions (2,nslits)')
+            spec_min_max = spec_min_max_tmp
+        else:
+            msgs.error('Invalid shapes for traces_min and traces_max')
+    else:
+        nslits = 1
+        traces = traces_in.reshape(traces_in.size, 1)
```

Comment:
Can also do: `traces_in.reshape(-1,1)`

Reply:
I guess I find the current version slightly more transparent.